compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -104,7 +104,7 @@ object DesugarEnums {
 
   /**  The following lists of definitions for an enum type E and known value cases e_0, ..., e_n:
    *
-   *   private val $values = Array[E](e_0,...,e_n)(ClassTag[E](classOf[E]))
+   *   private val $values = Array[E](e_0,...,e_n)(ClassTag[E](classOf[E])): @unchecked
```

Comment:
Please add a comment explaining the interaction with the init checker. Is this a known limitation of the init checker? Is it something that can be fixed and if so is there an issue for it?

Reply:
The issue is that the init checker will not try to prove that `new E {...}` is safe for performance reasons, so all enum cases that call `$new` creator method can't be checked. The init checker however knows that selecting from `E` companion object must be safe because it is already constructed, so we can either select from `this` which means `$values` does not have to be lazy, or we have to select from `E` which is under construction so it has to be lazy.

Reply:
cc @liufengyun

Reply:
we also have this comment https://github.com/lampepfl/dotty/pull/9665#issuecomment-682536402

Reply:
OK, as long as this is all properly documented and referenced in the code, I'm happy :).

Reply:
In the long run, we might introduce _type annotations_ and/or eagerly check _small classes_, both could be helpful to avoid `@unchecked` in such cases. Initially, we want to keep the checker stupid to see how it works in the wild and make sure the core algorithm is reliable.

Reply:
Maybe an issue should be opened for that (either here or on dotty-feature-requests) so that Jamie can link to it in the code?

Reply:
I opened https://github.com/lampepfl/dotty-feature-requests/issues/135